stravalib/model.py

```diff
@@ -59,26 +59,49 @@
 from stravalib.strava_model import (
     Split,
     SportType,
+    SummaryClub,
     SummaryPRSegmentEffort,
     SummarySegmentEffort,
     TimedZoneRange,
 )
 
+if TYPE_CHECKING:
+    from stravalib.client import BatchedResultsIterator
+
 LOGGER = logging.getLogger(__name__)
 
 
-def lazy_property(fn):
+# TODO: stravalib/model.py:71: error: Please use "Callable[[<parameters>], <return type>]"  [misc]
```

Comment:
i need help typing this Callable - i can't seem to get this one right - i understand that it's a function but i don't understand how to know what the parameters are to type - @enadeau any suggestions??

Reply:
i suspect this should be Callable[Any], Any] ... thank you mypy docs :) is that right?

Reply:
Sorry for not responding before - too busy at the moment. I think your solution will work, but maybe it can/should be more precise (which is not my expertise). @enadeau, Any thoughts on this? Will the correct typing be inferred from the returns of the decorated methods?

Reply:
Yes a decorator is in fact a function that takes a function and return a function. Mypy docs has a good section on it https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators

We can probably be quite precise with the types that go in. I'd need to look at the code a bit. I'll try to come back to this quickly

Reply:
ok thank you both! and @jsamoocha no worries we are all busy and i get that. i'm just enjoying learning a bunch here from you both. i pushed some cleanup just now. don't worry about the tests failing now - i think it's related to some typing issues / errors that i have here. we can just chip away at this until mypy is happy!!